sotodlib/preprocess/processes.py

```diff
@@ -1621,6 +1658,63 @@ def process(self, aman, proc_aman):
         if self.process_cfgs:
             pointing_model.apply_pointing_model(aman)
 
+class BadSubscanFlags(_Preprocess):
+    """Identifies and flags bad subscans.
+
+      Example config block::
+
+        - name : "noisy_subscan_flags"
+            stats_name: tod_stats # optional
+            calc: 
+                nstd_lim: 5.0 
+                merge: False
+            save: True
+            select: True
+    
+    .. autofunction:: sotodlib.tod_ops.flags.get_badsubscan_flags
+    """
+    name = "noisy_subscan_flags"
+
+    def __init__(self, step_cfgs):
+        self.stats_name = step_cfgs.get('stats_name', 'tod_stats')
+        super().__init__(step_cfgs)
+    
+    def calc_and_save(self, aman, proc_aman):
+        if 'flags' not in proc_aman._fields:
+            from sotodlib.core import FlagManager
+            proc_aman.wrap('flags', FlagManager.for_tod(proc_aman))
+        proc_aman.flags.wrap("left_scan", aman.flags.left_scan)
+        proc_aman.flags.wrap("right_scan", aman.flags.right_scan)
+
+        subscan_stats_T = proc_aman[self.stats_name+"_T"]
+        subscan_stats_Q = proc_aman[self.stats_name+"_Q"]
+        subscan_stats_U = proc_aman[self.stats_name+"_U"]
```

Comment:
So this requires that you run `GetStats` three times wrapping in the same name but appending `_T`, `_Q`, and `_U` to the end? That's a pretty specific prerequisite. Plus I think 90% of the time we are running `GetStats` it'll be to achieve what `BadSubscanFlags` is achieving (probably the other 10% of the time is getting some stats on the full TOD or PSD I guess). So why don't we instead check if `self.stats_name+"_T"` exists and if not call tod_ops.flags.get_stats to get the info you need.